modules/rating/src/main/glicko2/Result.scala

```diff
@@ -2,50 +2,49 @@ package lila.rating.glicko2
 
 trait Result:
 
-  def getScore(player: Rating): Double
+  val first: Rating
+  val second: Rating
 
-  def getOpponent(player: Rating): Rating
+  def getAdvantage(advantage: Double, player: Rating): Double =
+    if player == first then advantage / 2.0d else -advantage / 2.0d
```

Comment:
It's a bit insane that players are identified by their exact `Rating`. Not introduced by this PR, though.

Reply:
Indeed, there is a dilemma when two players have the exact same `Rating`.

Reply:
yeah that comes from https://github.com/goochjs/glicko2. We could fix it.

Reply:
I'll try to fix it... but also, there's a case of dependency inversion since `RatingCalculator` shouldn't need to inspect `RatingPeriodResults` to identify what players, puzzles, tutor-exercises, etc. need ratings updated.

Trait `RatingPeriodResults` should be replaceable by `List[Result]`, because each consumer of `RatingCalcuator#updateRatings` already has that context, especially because we're only rating games and puzzles serially (one at a time) anyway (except for `modules/tutor/src/test/GlickoTest.scala` which rates 2 results concurrently).

Reply:
in any case this PR is not the time and place to fix it.